modules/claims_api/app/sidekiq/claims_api/va_notify_job.rb

```diff
@@ -0,0 +1,184 @@
+      {
+        recipient_identifier: @icn_for_vanotify,
+        personalisation: {
+          first_name: value_or_default_for_field(claimant_first_name(poa)),
+          org_name: value_or_default_for_field(org.name),
+          address: build_org_address(org),
+          city: value_or_default_for_field(org.city),
+          state: value_or_default_for_field(org.state_code),
+          zip: value_or_default_for_field(org_zip(org)),
+          phone: value_or_default_for_field(org.phone)
+        },
+        template_id: Settings.claims_api.vanotify.service_organization_template_id
+      }
+    end
+
+    def find_org(poa, form_number)
+      poa_code = poa_code_from_form(form_number, poa)
+      ::Veteran::Service::Organization.find_by(poa: poa_code)
+    end
+
+    def veteran_icn_identfier(poa)
+      poa.source_data&.dig('source_data', 'icn')
+    end
+
+    def rep_registration_number(poa)
+      poa.form_data.dig('representative', 'registrationNumber')
+    end
+
+    def rep_phone(rep)
+      # This field was added to adjust the values for phone numbers
+      # This should the the reps specific phone number
+      if rep.phone_number.present?
+        rep.phone_number
+      # This might be the phone number for the organization
+      # the rep works for
+      elsif rep.phone.present?
+        rep.phone
+      else
+        ''
+      end
+    end
+
+    def rep_zip(poa)
+      first_five = poa.form_data.dig('representative', 'address', 'zipCode')
+      last_four = poa.form_data.dig('representative', 'address', 'zipCodeSuffix')
+      format_zip_values(first_five, last_four)
+    end
+
+    def org_zip(org)
+      zip = org.zip_code
+      suffx = org.zip_suffix
+      format_zip_values(zip, suffx)
+    end
+
+    def format_zip_values(first, last)
+      # neither are required for either form
+      if first && last
+        "#{first}-#{last}"
+      elsif first && !last
+        first
+      elsif !first && last
+        last
+      end
+    end
+
+    def value_or_default_for_field(field)
+      field || ''
+    end
+
+    def poa_form_data(poa)
+      poa&.form_data
+    end
+
+    def build_ind_poa_address(poa)
+      form_data = poa&.form_data
+      address_line1 = form_data&.dig('representative', 'address', 'addressLine1')
+      address_line2 = form_data&.dig('representative', 'address', 'addressLine2')
+      address_line3 = form_data&.dig('representative', 'address', 'addressLine3')
+
+      build_address(address_line1, address_line2, address_line3)
+    end
+
+    def build_org_address(org)
+      address_line1 = org.address_line1
+      address_line2 = org.address_line2
+      address_line3 = org.address_line3
+
+      build_address(address_line1, address_line2, address_line3)
+    end
+
+    def build_address(line1, line2, line3)
+      address = ''.dup
+      address << line1 ? line1.to_s : ''
+      address << "\n #{line2}" if line2.present?
+      address << "\n #{line3}" if line3.present?
+
+      address
+    end
```

Comment:
similar premise as above - maybe a compact blank and join here? save a few lines.

Reply:
same